settings: Deprecate base64 in fs backed settings

de-nordic opened the pull request:
Encoding of values with use of base64 has been marked as deprecated and
will be removed in future releases.

GH Issue #18927

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nvlsianpu reviewed (changes requested):
I believe this PR should make RAW all new write to setting storage. Support of bas64 writes should be left behind here as base64 readout is the deprecation steep. cc @carlescufi

de-nordic commented on subsys/settings/src/settings_file.c:
```diff
@@ -146,7 +153,18 @@ static int settings_file_load_priv(struct settings_store *cs, line_load_cb cb,
+		 * check for it and mark its position if found.
+		 */
+		if (entry_ctx.b64_marker == false &&
+		    strncmp(name, NOB64_MARKER, name_len - 1) == 0) {
```
> 
> 
> I believe this PR should make RAW all new write to setting storage. Support of bas64 writes should be left behind here as base64 readout is the deprecation steep. cc @carlescufi

Write is supposed to be completely disabled. I am looking for marker only to know it it has already been placed or it should be written before actual line is written.

de-nordic replied:
> 
> 
> need to compare string length in case `name_len - 1 < sizeof(NOB64_MAKER) - 1`

Yes indeed. I missed that...

carlescufi commented:
@nvlsianpu please re-review

nvlsianpu commented:
Patch looks good although still need to add deprecation BASE64 write encoding. I see two possible way:
1) remove CONFIG_SETTINGS_USE_BASE64 now and ensure backward compatibility using CONFIG_SETTINGS_DEPRECATE_USE_BASE64.
2) deprecate base64 write encoding (so SETTINGS_USE_BASE64 Kconfig) flowing https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#deprecated
CONFIG_SETTINGS_DEPRECATE_USE_BASE64 will ensure backward compatibility after CONFIG_SETTINGS_USE_BASE64 will disappear. 

I have doubt on what to chose because this patch deprecates internal feature behavior - not the API.

de-nordic committed:
settings: Deprecate base64 in fs backed settings

Encoding of values with use of base64 has been marked as deprecated and
will be removed in future releases.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nvlsianpu reviewed (approved):
just fix the typo
